spec/models/user_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validates' do
+    subject { User.new(name: 'Tom', photo: 'Tom.png', bio: 'Teacher from Mexico.') }
+
+    before { subject.save }
+
+    it 'should have a name' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'return all the posts for a user' do
+      expect(User.count).to eql 0
+    end
+  end
+
+  describe 'Should test recent post method' do
+    before { 4.times { |post_nb| Post.create(author_id: subject.id, title: "This is post #{post_nb}") } }
+
+    it 'User should have three recent posts' do
+      expect(subject.recent_posts).to eq(subject.posts.last(3))
+    end
+  end
+end
```

Comment:
- [ ] I could not successfully create a database after cloning your project using the `rails db:create` command. I got an error `NameError: undefined local variable or method 'create' for #<ActionDispatch::Routing::Mapper:0x0000028f101a79e8>`. Maybe there is a configuration you did that makes it difficult to create a database, if any? kindly add detailed instructions on how to set up your project's database. Because of this inability to create a database, I couldn't run your tests.

Reply:
- [ ] I am having the exact same error as the previous reviewer it is likely that there is something wrong with the way the project is set up on your side. you should try to find the cause of why the rails application is complaining that `create` is undefined. looking for places where `create` method is being called may give a hint as to where the problem is coming from. I suggest you reclone your project in a different directory and see if you will see the same error that we are getting. remember that your project should be set up in a way that someone can clone it and install the dependencies and it should run correctly.

Reply:
How do I solve a issue that I am not facing from my end? Probably a screenshot would make me understand the kind of error you facing.

Reply:
I understand what you mean. the error I get is the same one that @julie-ify got. my suggestion to you is that you can quickly reclone your project in a new directory and see if you will face the issue we are facing with a new project.

Reply:
![image](https://user-images.githubusercontent.com/59098116/161379761-c3907048-5999-4db1-8a85-06ca67f1886d.png)

Attached is the image of my test ran from another system,
I don't know the issue you guys are facing and I don't know how to tackle it, spent the whole night trying to fix it, however, any Reviewer should take note of these comments.

secondly, regarding the counter, I was told to implement, after exhausting all available resources the counters are private so I was unable to write the test for the counter if you think there is away, I am much open to learning, all my colleagues I reached out to didn't implement it.

Reply:
Hi @zemola, I'm able to set up your database now and successfully run all the tests and they passed, However, I tried starting your project server `rails s` but I got the error from the screenshot below. I don't know how you created your rails application that made it different from others and it has been attracting errors. Kindly fix all these errors for your future projects so you don't have to go through this stressful process again. I will approve your project because your tests are passing and you've made the methods private. Have a nice weekend. 🤝


See the error from running rails server;

<img width="469" alt="Code_meTdGoM11N" src="https://user-images.githubusercontent.com/81468675/161382785-b15c3888-5a9f-4d2a-9b24-70a0c2dcee9c.png">